Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.cpp

```diff
@@ -27,7 +27,40 @@
 #include <wtf/NeverDestroyed.h>
 #include <wtf/glib/GUniquePtr.h>
 
-static HashSet<GObject*> s_watchedObjects;
+class Watcher {
+public:
+    ~Watcher()
+    {
+        checkLeaks();
+    }
+
+    void assertObjectIsDeletedWhenTestFinishes(GObject* object)
+    {
+        m_watchedObjects.add(object);
+        g_object_weak_ref(object, [](gpointer data, GObject* finalizedObject) {
+            Watcher* watcher = static_cast<Watcher*>(data);
+            watcher->m_watchedObjects.remove(finalizedObject);
```

Comment:
Optional: I think the standard name for this parameter is slightly nicer:

```suggestion
        g_object_weak_ref(object, [](gpointer data, GObject* whereTheObjectWas) {
            Watcher* watcher = static_cast<Watcher*>(data);
            watcher->m_watchedObjects.remove(whereTheObjectWas);
```

Reply:
Done.